plugins/org.jkiss.dbeaver.ext.oracle/src/org/jkiss/dbeaver/ext/oracle/model/OracleUtils.java

```diff
@@ -171,6 +176,33 @@ public static String getDDL(
         }
     }
 
+    private enum DBMSMetaDependentObjectType {
+        INDEX,
+        CONSTRAINT,
+        REF_CONSTRAINT,
+        TRIGGER
+    }
+
+    private static String invokeDBMSMetadataGetDependentDDL(JDBCSession session, OracleSchema schema, OracleTableBase object, DBMSMetaDependentObjectType dependentObjectType) {
+        StringBuilder sb = new StringBuilder();
+        try (JDBCPreparedStatement dbStat = session.prepareStatement(
+                "SELECT DBMS_METADATA.GET_DEPENDENT_DDL('" + dependentObjectType.name() + "',?" + (schema == null ? "" : ",?") + ") TXT FROM DUAL")) {
+            dbStat.setString(1, object.getName());
+            if (schema != null) {
+                dbStat.setString(2, schema.getName());
+            }
+            try (JDBCResultSet dbResult = dbStat.executeQuery()) {
+                if (dbResult.next()) {
+                    sb.append("\n\n").append(dbResult.getString(1).trim());
+                }
+            }
+        } catch (Exception e) {
+            // No dependent index DDL or something went wrong
+            log.debug("Error reading dependent index DDL", e);

```

Comment:
Perhaps the caller should handle the exceptional situation themselves? It will allow providing human-readable error messages. Otherwise, at least dump some of the method's parameters.

Reply:
It used to work this way before. I hope it was done for some purpose. Not sure that it's good to change this behavior.